sources/packages/backend/apps/api/src/route-controllers/student/student.external.controller.ts

```diff
@@ -0,0 +1,51 @@
+import { Body, Controller, NotFoundException, Post } from "@nestjs/common";
+import { ApiNotFoundResponse, ApiTags } from "@nestjs/swagger";
+import { ClientTypeBaseRoute } from "../../types";
+import { AuthorizedParties } from "../../auth/authorized-parties.enum";
+import {
+  AllowAuthorizedParty,
+  RequiresUserAccount,
+} from "../../auth/decorators";
+import BaseController from "../BaseController";
+import { StudentService } from "@sims/integrations/services";
+import {
+  ExternalSearchStudentAPIInDTO,
+  StudentDetailsAPIOutDTO,
+} from "./models/student.dto";
+
+/**
+ * Student controller for external client.
+ */
+@RequiresUserAccount(false)
+@AllowAuthorizedParty(AuthorizedParties.external)
+@Controller("student")
+@ApiTags(`${ClientTypeBaseRoute.External}-student`)
+export class StudentExternalController extends BaseController {
+  constructor(private readonly studentService: StudentService) {
+    super();
+  }
+
+  /**
+   * Searches for student student details.
+   * @param payload payload with sin to retrieve the student details.
+   * @returns student details.
+   */
+  @Post()
+  @ApiNotFoundResponse({ description: "Student not found." })
```

Comment:
Please decorate the endpoint to ensure the right HTTP response code, since nothing is being created.
```
@HttpCode(HttpStatus.OK)
```